File: LeetCode/BinaryInOrder.py
# Ques :- 94 ... Inorder traversal of Binary Tree.

class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        result = []

        def inorder(node):
            if node is None:
                return

            inorder(node.left)
            result.append(node.val)
            inorder(node.right)

        inorder(root)
        return result

root = TreeNode(1, TreeNode(2, TreeNode(4), TreeNode(5)), TreeNode(3))
print(Solution().inorderTraversal(root))

File: BasicToAdvance/Java/LeetCode/BinaryInorder.java
public class BinaryInorder {
    static class TreeNode {
        int val;
        TreeNode right;
        TreeNode left;

        TreeNode(int val) {
            this.val = val;
        }
    }

    static void BinInorder(TreeNode node, java.util.List<Integer> result) {
        if (node == null) {
            return;
        }

        BinInorder(node.left, result);
        result.add(node.val);
        BinInorder(node.right, result);
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.right = new TreeNode(2);
        root.right.left = new TreeNode(3);

        java.util.List<Integer> result = new java.util.ArrayList<>();
        BinInorder(root, result);

        System.out.println(result);
    }
}
